### ISSUE TITLE
Should we rename <MethodName>Reply type to something else?

### ISSUE THREAD

**Issue by meling** [relab/gorums#33]
The following is just a small excerpt from a longer discussion in [relab/gorums-dev/#29](https://github.com/relab/gorums-dev/issues/29). The below description captures the main outstanding issue that haven't been decided yet. Namely what to name the reply type, when it is different from type specified in the proto file, because we need to return a more sophisticated data structure, for call types such as correctables and futures.

**Below text is copied from the old issue:**
Note that, I'm not happy with the suffix `Reply` to the `ReadCorrectableReply` which comes from the `ReadCorrectable` method name.
```
sm.TypeName = sm.MethodName + "Reply"
```
Ignoring the fact that in the current example, I've named the method `ReadCorrectable`, perhaps it would be better to name the different reply types with their respective names, e.g. `Correctable`. That is, for a `ReadCorrectable` method, the reply type would become `ReadCorrectableCorrectable`.

This looks ugly for our example case, but if our method name is just `Read`, then the return type becomes `ReadCorrectable` and `ReadFuture` and so on.

Another alternative (that was used in a previous implementation) is to just use the method name as the name for the reply type.

Anyone have any input on what is the preferred choice on this naming??

**Comment by meling** [comment]
Would it make sense to use
```go
sm.TypeName = sm.CustomRespName + "Wrapper"
```
Note that `sm.CustomRespName` is set to `sm.RespName` if `sm.CustomRespName` is the empty string.

This would give us reply types such as `StateWrapper` and `ReadRequestWrapper`.

However, I'm not sure if this unique, e.g. if multiple methods return `State`.

**Comment by meling** [comment]
Closed by 35059f7
